src/components/WwwFrame/DisclaimersContentful.vue

```diff
@@ -0,0 +1,131 @@
+<template>
+	<div class="row">
+		<ol id="disclaimers" class="text-left">
+			<li
+				v-for="(disclaimer, index) in fullyBuiltDisclaimerText"
+				:key="index"
+				v-html="disclaimer"
+			>
+			</li>
+		</ol>
+	</div>
+</template>
+
+<script>
+import _get from 'lodash/get';
+import gql from 'graphql-tag';
+import { settingEnabled, settingWithinDateRange } from '@/util/settingsUtils';
+import { documentToHtmlString } from '~/@contentful/rich-text-html-renderer';
+
+const bannerQuery = gql`query bannerQuery {
+	contentful {
+		entries(contentType: "uiSetting", contentKey: "ui-global-promo")
+	}
+}`;
+
+export default {
+	data() {
+		return {
+			disclaimerContent: [],
+		};
+	},
+	inject: ['apollo', 'cookieStore'],
+	apollo: {
+		query: bannerQuery,
+		preFetch: true,
+		result({ data }) {
+			this.disclaimerContent = [];
+			// gather contentful content and the uiSetting key ui-global-promo
+			const contentfulContent = data?.contentful?.entries?.items ?? [];
+			const uiGlobalPromoSetting = contentfulContent.find(item => item.fields.key === 'ui-global-promo');
+
+			// exit if missing setting or fields
+			if (!uiGlobalPromoSetting || !uiGlobalPromoSetting.fields) {
+				return false;
+			}
+			// uiGlobalPromoSetting can contain an array of different banners with
+			// different start/end dates first determine if setting is enabled.
+			const isGlobalSettingEnabled = settingEnabled(
+				uiGlobalPromoSetting.fields,
+				'active',
+				'startDate',
+				'endDate'
+			);
+
+			// if setting is enabled determine which banner to display
+			if (isGlobalSettingEnabled) {
+				const activePromoBanner = uiGlobalPromoSetting.fields.content.find(promoContent => {
+					return settingEnabled(
+						promoContent.fields,
+						'active',
+						'startDate',
+						'endDate'
+					);
+				});
+
+				// gather all inactive promo banners by their start and end dates
+				const inactivePromoBanners = uiGlobalPromoSetting.fields.content.filter(promoContent => {
+					const hiddenUrls = promoContent?.fields?.hiddenUrls ?? [];
+					// check hiddenUrl for display of disclaimers
+					if (hiddenUrls.includes(this.$route.path)) {
+						return false;
+					}
+					if (promoContent.fields.active) {
+						return false;
+					}
+					return settingWithinDateRange(
+						promoContent.fields,
+						'startDate',
+						'endDate'
+					);
+				});
+
+				if (activePromoBanner) {
+					// check for visibility based on current route and hiddenUrls field
+					const hiddenUrls = _get(activePromoBanner, 'fields.hiddenUrls', []);
+					if (hiddenUrls.includes(this.$route.path)) {
+						return false;
+					}
+
+					// check for visibility on promo session override
+					const showForPromo = _get(activePromoBanner, 'fields.showForPromo', false);
+					if (this.hasPromoSession && !showForPromo) {
```

Comment:
`this.hasPromoSession` isn't present in this component so we may want to add that check or pass down that state.

Reply:
As of this time, this is just never going to trigger, so it's not a blocker, but something we should get in place at some point soon.